pdns/ws-auth.cc

```diff
@@ -1510,6 +1687,16 @@ static void patchZone(HttpRequest* req, HttpResponse* resp) {
     di.backend->abortTransaction();
     throw;
   }
+
+  DNSSECKeeper dk;
+  string api_rectify;
+  di.backend->getDomainMetadataOne(zonename, "API-RECTIFY", api_rectify);
+  if (dk.isSecuredZone(zonename) && !dk.isPresigned(zonename) && api_rectify == "1") {
+    string error_msg = "";
+    if (!dk.rectifyZone(zonename, error_msg))
+      throw ApiException("Failed to rectify '" + zonename.toString() + "' " + error_msg);
+  }
+
   di.backend->commitTransaction();
```

Comment:
dk.rectifyZone calls commitTransaction too, right? Maybe dk.rectifyZone shouldn't deal with backend lookups and transactions at all.

Reply:
That would mean that all callers are responsible for this? I would prefer dk.rectifyZone() to do some sanity checking and perform adding the ENTs in a safe way

Reply:
Probably. But having the transaction being committed underneath you is also super surprising!

Reply:
Probably. But it'd be great if the API could modify *and* rectify the zone in one TX, so no unrectified stuff is ever served by the DNS parts...

Reply:
that is a sexy idea......

Reply:
Will be implemented